worker/machiner/manifold.go

```diff
@@ -91,7 +91,13 @@ func newWorker(a agent.Agent, apiCaller base.APICaller) (worker.Worker, error) {
 		Tag:                          tag.(names.MachineTag),
 		ClearMachineAddressesOnStart: ignoreMachineAddresses,
 		NotifyMachineDead: func() error {
-			return agent.SetCanUninstall(a)
+			// achilleasa: uninstalling the agent is probably not
```

Comment:
We've decided not to include this one as it has "normal use case" implications. So lets back out this change and then land your patch into 2.5 and beyond.

And then we'll do a bigger spec around "if we never want to uninstall ourselves how do we recover a manual machine", and when we have the answer worked out, we can land this part of the change.

Also, fwiw, we should just set NotifyMachineDead: nil, not a func() that does nothing.

Reply:
Totally agree about setting it to nil. I just left it there for the commented line to be visible.

I will drop this commit and proceed as you suggest.